eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/ManagementInputLibrary.cs

```diff
@@ -166,5 +167,45 @@ ResourceMetadata BuildResourceMetadata(InputDecoratorInfo decorator)
                     parentResource);
             }
         }
+
+        internal static bool IsResourceUpdateModel(InputModelType model)
+        {
+            const string ResourceUpdateModelId = KnownManagementTypes.ResourceUpdateModelId;
+
+            var currentModel = model;
+            while (currentModel != null)
+            {
+                if (currentModel.CrossLanguageDefinitionId.Equals(ResourceUpdateModelId, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+                currentModel = currentModel.BaseModel;
+            }
+
+            return false;
+        }
+
+        internal static string FindEnclosingResourceNameForResourceUpdateModel(InputModelType model)
+        {
+            var inputLibrary = ManagementClientGenerator.Instance.InputLibrary;
+            foreach (var client in inputLibrary.InputNamespace.Clients)
```

Comment:
consider to populate the information during
https://github.com/Azure/azure-sdk-for-net/blob/bb45bbd6e224325d8edb85a4beae1741bf739d24/eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/ManagementInputLibrary.cs#L41
to avoid the duplicated looping